great_expectations/core/data_context_key.py

```diff
@@ -80,19 +80,17 @@ def from_fixed_length_tuple(cls, tuple_):
 class DataContextVariableKey(DataContextKey):
     def __init__(
         self,
-        resource_type: "DataContextVariableSchema",  # noqa: F821
         resource_name: Optional[str] = None,
     ) -> None:
-        self._resource_type = resource_type
         self._resource_name = resource_name
 
-    def to_tuple(self) -> Tuple[str, Optional[str]]:
+    def to_tuple(self) -> Tuple[Optional[str]]:
```

Comment:
Possibly look into `NamedInlineStoreBackend` (a way to separate out nested get/set behavior for items like datasources). `plugins_directory` and other simple variables would still utilize the existing backend.